### ISSUE TITLE
Unexpected warning for Flexbox support in IE10, 11

### ISSUE THREAD

**Issue by shadeed** [stylelint/stylelint#1474]
Hello Guys,

Thanks a lot for your time spent on this useful plugin.

There is a bug where `no-unsupported-browser-features` rule shows that Flexbox is not supported in IE10 and IE11.

```css
.foo {
  display: flex;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "no-unsupported-browser-features": [ true, {
      "browsers": "> 1%, last 2 versions, ie >= 9"
   } ]
}
```

> Which version of stylelint are you using?

`6.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Grunt and postcss

> What did you expect to happen?

It should warn for Flexbox support only in IE9 and below.

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
Unexpected browser feature "flexbox" is not supported by IE 9,10,11, IE Mobile 10, UC Browser for Android 9.9 (no-unsupported-browser-features) [stylelint]
```
Am I missing anything or this is expected? 

Thanks again,

**Comment by m-allanson** [comment]
Hi @shadeed, I think this is caused by caniuse listing [IE 10 & 11 as having 'partial' support for flexbox](http://caniuse.com/#search=flexbox). `no-unsupported-browser-features` relies on the caniuse database. 

As a workaround there's an [optional ignore parameter for this rule](https://github.com/stylelint/stylelint/tree/master/src/rules/no-unsupported-browser-features#ignore--array-of-features-to-ignore-). You can find the list of ignorable feature names over in the doiuse project: https://github.com/anandthakker/doiuse/blob/master/data/features.js#L93

**Comment by ntwb** [comment]
#1401 is also affected by this same issue with the `doiuse` library,  https://github.com/anandthakker/doiuse/issues/15

**Comment by m-allanson** [comment]
Should we update the `no-unsupported-browser-features` docs?

- Add a notice that partially supported features will trigger a warning
- Add a link to the doiuse list of ignorable feature names

**Comment by jeddy3** [comment]
> Should we update the no-unsupported-browser-features docs? And then close this issue in favour of #1401

Sounds great to me.

**Comment by jeddy3** [comment]
Additionally we should update the rule message to read:

`Unexpected browser feature "flexbox" is not fully supported by x, y and z`

**Comment by shadeed** [comment]
@jeddy3 can you please explain more about how I can help out in `doiuse`?

Regarding updating the rule message, I see that this is perfect and will do the job.

**Comment by jeddy3** [comment]
@shadeed [`doiuse`](https://github.com/anandthakker/doiuse/) is the library we use behind the scenes in the `no-unsupported-browser-features` rule. It currently only checks for *full* support, rather than distinguishing between *partial* and *full* support. The author of `doiuse` has a [open issue](https://github.com/anandthakker/doiuse/issues/15#issuecomment-107102958) about this limitation and is asking for help to improve the library. So, if you'd like our `no-unsupported-browser-features` rule to distinguish between *partial* and *full* support then please consider helping resolve this limitation in [`doiuse`](https://github.com/anandthakker/doiuse/).
